chore: Update issue template to GH form

dnys1 opened the pull request:
*Issue #, if available:*
The current bug report template on Github leaves a lot to be desired. Looking at recent issues, many people forego the template altogether and add their own input but this can have the effect of missing critical information.

*Description of changes:*

![Form](https://docs.github.com/assets/cb-61077/images/help/repository/sample-issue-form.png)

This introduces a migration to the [Issue Form](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) which other Amplify teams currently use. It provides a nice web form interface with proper input validation, text boxes, formatting, etc.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

HuiSF commented:
Also a quick check @dnys1 for the textarea where to paste command line output, will the final results be converted into `<details>` wrapped block? That would be helpful for reading and navigating the issue folding longer section by default. Thanks!

dnys1 commented:
> Also a quick check @dnys1 for the textarea where to paste command line output, will the final results be converted into `<details>` wrapped block? That would be helpful for reading and navigating the issue folding longer section by default. Thanks!

Not sure, but I've taken steps to reduce to length of these outputs:
1. Changed `flutter doctor -v` to just `flutter doctor`
2. Changed "paste pubspec.lock" to a built-in command which is concise (`flutter pub deps --style=compact`)

Both of these outputs are only about a paragraph
